### PR TITLE
HL-426: CSV export of application data

### PR DESCRIPTION
**Pull request by haavikko**
Includes HL-428: Finnish column headings for CSV data
Also includes HL-408: application cancellation reason.
Implementing CSV export this way also covers the functionality in
HL-425.

### CODE REVIEW

File: backend/benefit/applications/services/csv_export_base.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+        csv_string = self.get_csv_string()
+        with open(path, encoding=self.FILE_ENCODING, mode="w") as f:
+            f.write(csv_string)
+
+    def get_csv_string(self):
+        return self._make_csv(self.get_csv_lines())
+
+    def get_csv_lines(self):
+        lines = [self._get_header_row()]
+        for item in self.get_row_items():
```
Comment (ngocdh236) [inline backend/benefit/applications/services/csv_export_base.py:68]: Could you please check, is this `get_row_items` method missing..?

Comment (haavikko) [reply backend/benefit/applications/services/csv_export_base.py:68]: It's the responsibility of a derived class to provide that, like TalpaService and ApplicationsCsvService do

File: backend/benefit/applications/services/csv_export_base.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    return getter
+
+
+class CsvExportBase:
+
+    CSV_DELIMITER = ";"
+    FILE_ENCODING = "utf-8"
+
+    def _get_header_row(self):
+        return [col.heading for col in self.CSV_COLUMNS]
```
[the remark this answers is not in the document]
Comment (haavikko) [reply backend/benefit/applications/services/csv_export_base.py:56]: Same as above. I'll add a comment about this

File: backend/benefit/applications/services/csv_export_base.py (hunk 1 above)
Comment (ngocdh236) [reply backend/benefit/applications/services/csv_export_base.py:84]: Ah yes thank you. Maybe we can add these as abstract method / property..?

**Review by ngocdh236 — APPROVED** [review]

**Pull request merged**